docs/application/native/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,618 @@
+
+2. Add a service-component to the component-based Application:
+    ```
+    component_class_h service_component_add(component_class_h comp_class,
+            const char *component_id, void *user_data)
+    {
+        service_component_lifecycle_callback_s callback = {
+            .create = __service_component_create_cb,
+            .start_command = __service_component_start_command_cb,
+            .destroy = __service_component_destroy_cb,
+            .restore_content = __service_component_restore_content_cb,
+            .save_content = __service_component_save_content_cb,
+            .action = __service_component_action_cb,
+            .device_orientation_changed = __service_component_device_orientation_changed_cb,
+            .language_changed = __service_component_language_changed_cb,
+            .region_format_changed = __service_component_region_format_changed_cb,
+            .low_battery = __service_component_low_battery_cb,
+            .low_memory = __service_component_low_memory_cb,
+            .suspended_state_changed = __service_component_suspended_state_changed_cb,
+        };
+
+        return component_based_app_add_service_component(comp_class,
+                component_id, &callback, user_data);
+    }
+
+    component_class_h __app_create_cb(void *user_data)
+    {
+         component_class_h comp_class = NULL;
+         comp_class = frame_component_add(comp_class, "base-frame", NULL);
+         comp_class = service_component_add(comp_class, "base-service", NULL);
+        return comp_class;
+    }
+
+    void __app_terminate_cb(void *user_data)
+    {
+    }
+
+    int main(int argc, char** argv)
+    {
+        int ret;
+        component_based_app_lifecycle_callback_s callback = {
+            .create = __app_create_cb,
+            .terminate = __app_terminate_cb
+        };
+
+        ret = component_based_app_main(argc, argv, &callback, NULL);
+        if (ret != APP_ERROR_NONE)
+            return ret;
+
+        return 0;
+    }
+    ```
+
+<a name="register_action"></a>
+## Managing Actions
+
+The component-based Application can register actions and receive action event. You can send an action event to another application using app-control.
+
+1. The app-control declaration in the manifest file
+    ```
+    <component-based-application appid="org.tizen.base-component" exec="@BINDIR@/base-component" nodisplay="false" multiple="false" type="capp">
+        <label>Base-component Application</label>
+        <icon>@DESKTOP_ICON@</icon>
+        <app-control id="dial">
+            <operation name="http://tizen.org/appcontrol/operation/dial"/>
+            <mime name="*"/>
+        </app-control>
+        <app-control id="dial-for-excel">
+            <operation name="http://tizen.org/appcontrol/operation/dial"/>
+            <mime name="application/vnd.ms-excel"/>
+        </app-control>
+    ```
+
+2.  Register an action
+
+    To receive action events, each component instance must register the action.
```

Comment:
How about changing the last "action" to "actions"?

~ must register the actions.

Reply:
As the heading is "Register an action", then, "To receive action events, each component instance must register the action." is correct.
That means, for one action, we have multiple action events.
As suggested, if we change to "must register the actions." Then the meaning will slightly change.
I think we can wait for @gotoextreme to clarify this.